chain/indexer/src/streamer/utils.rs

```diff
@@ -25,6 +25,7 @@ pub(crate) async fn convert_transactions_sir_into_local_receipts(
     let local_receipts: Vec<views::ReceiptView> =
         txs.into_iter()
             .map(|tx| {
+                assert_eq!(tx.transaction.signer_id, tx.transaction.receiver_id);
```

Comment:
note to reviewer: I could not validate that this condition should always hold.  As the previous code was always setting `sender_is_receiver` to `true`, I inserted this assertion here.  

It might also not be a bad assertion given the name of the function.